components/mission_protection_system/src/posix_main.c

```diff
@@ -220,7 +318,227 @@ int read_mps_command(struct mps_command *cmd) {
     clean_exit(0);
 
   if (linelen < 0)
+    return NULL;
+
+  // required because we don't have a predicate for returning a partially-initialized string
+  /*$ apply ViewShift_Block_u8(line, array_shift(line, (u64)linelen), (u64)linelen, linecap-(u64)linelen); $*/
+  memset(&line[linelen], 0, linecap-linelen);
+  /*$ apply UnViewShift_Owned_u8(line, array_shift(line, (u64)linelen), (u64)linelen, linecap-(u64)linelen); $*/
+  /*$ apply JoinSlice_Owned_u8(line, 0u64, linecap, (u64)linelen); $*/
+  return line;
+}
+
+void setup_mps_socket(void)
+/*$
+  accesses mps_cmd_fd;
+$*/
+{
+  int server_socket_fd;
+  struct sockaddr_un sockaddr_un = {0};
+  int return_value;
+  char *socket_address = getenv("MPS_SOCKET");
+  if (socket_address == NULL) {
+    return;
+  }
+
+  server_socket_fd = socket( AF_UNIX, SOCK_STREAM, 0 );
+  if ( server_socket_fd == -1 ) {
+    DEBUG_PRINTF(("failed to create socket %d", errno));
+    clean_exit(1);
+  }
+
+  remove( socket_address );
+
+  sockaddr_un.sun_family = AF_UNIX;
+  strcpy( sockaddr_un.sun_path, socket_address );
+  struct sockaddr_un saddr = sockaddr_un;
+
+  return_value =
+    bind(
+      server_socket_fd,
+       (const struct sockaddr*)(struct sockaddr*)&saddr,
+      sizeof( struct sockaddr_un ) );
+
+  if ( return_value == -1 ) {
+    DEBUG_PRINTF(("failed to bind the mps socket %d", errno));
+    clean_exit(1);
+  }
+
+  if ( listen( server_socket_fd, 4096 ) == -1 ) {
+    DEBUG_PRINTF(("failed to listen to the mps socket %d", errno));
+    clean_exit(1);
+  }
+
+  int accept_socket_fd = 0;
+
+  while (1)
+  /* $ inv
+    true;
+  $*/
+  {
+    accept_socket_fd = accept( server_socket_fd, NULL, NULL );
+    if ( accept_socket_fd == -1 ) {
+      DEBUG_PRINTF(("failed to accept the mps socket %d", errno));
+      clean_exit(1);
+    }
+
+    if ( accept_socket_fd > 0 ) { /* client is connected */
+      mps_cmd_fd = accept_socket_fd;
+      break;
+    }
+  }
+}
+
+#if 0
```

Comment:
`#if 0`? Not sure what is the purpose of this code?

Reply:
It was the realloc version of read_line_socket, I've deleted it